### PR TITLE
Add bytes sent and received metrics

### PR DESCRIPTION
**Pull request by ymesika**
PR for #310 

Adding support for the `istio_tcp_received_bytes_total` and `istio_tcp_sent_bytes_total` with their labels.
Currently only supports relay without zero copy as this hides the information about transferred bytes.

### CODE REVIEW

File: src/proxy/outbound.rs [hunk 1]
```diff
@@ -260,9 +264,18 @@ impl OutboundConnection {
                 // Create a TCP connection to upstream
                 let mut outbound = TcpStream::connect(req.gateway).await?;
                 // Proxying data between downstrean and upstream
-                relay(&mut stream, &mut outbound).await?;
-
-                Ok(())
+                match relay(&mut stream, &mut outbound, self.cfg.zero_copy_enabled).await {
```
Comment (howardjohn) [inline src/proxy/outbound.rs:267]: We will need this in other places as well, but this is a good start

**Review by howardjohn — APPROVED** [review]

File: src/proxy/outbound.rs [hunk 2]
```diff
@@ -260,9 +264,18 @@ impl OutboundConnection {
...
+                    // Connection closed with count of bytes transferred between streams
+                    Ok(Some(r)) => {
```
Comment (howardjohn) [inline src/proxy/outbound.rs:269]:
```suggestion
                    Ok(Some((sent, recv))) => {
```

instead of r.0 and r.1 is more clear

**Comment by howardjohn** [comment]
I added a hold to prevent auto-merge since there were some nits, feel free to remove

File: src/metrics/traffic.rs [hunk 3]
```diff
@@ -100,6 +102,10 @@ impl<T: Encode> Encode for DefaultedUnknown<T> {
 
 pub struct ConnectionClose<'a>(&'a ConnectionOpen);
 
+pub struct ReceivedBytes<'a>(pub &'a ConnectionOpen);
```
[the remark this answers is not in the document]
Comment (ymesika) [reply src/metrics/traffic.rs:105]: Without the pub I get an error that the constructor is private when creating an instance here: https://github.com/istio/ztunnel/blob/ab5b28b4ee74f8063ee11050c8a636f392fc360b/src/proxy/outbound.rs#L175

Not sure how to overcome it without pub.

**Pull request merged**